src/pages/filter/ui/enum-panel.tsx

```diff
@@ -1,76 +1,160 @@
-      if (checked) {
-        filterStore.addSelectedFilters({
-          group: filterStore.selectedGroupItem.vgroup,
-          groupItemName: filterStore.selectedGroupItem.name,
-          variant,
-        })
-
-        const conditionIndex = datasetStore.conditions.findIndex(
-          item => item[1] === filterStore.selectedGroupItem.name,
-        )
-
-        const enumValues =
-          conditionIndex === -1
-            ? [variant[0]]
-            : [...datasetStore.conditions[conditionIndex][3], variant[0]]
-
-        datasetStore.setConditionsAsync([
-          [
-            FilterKindEnum.Enum,
-            filterStore.selectedGroupItem.name,
-            '',
-            enumValues,
-          ],
-        ])
-
-        datasetStore.fetchWsListAsync()
-      } else {
-        filterStore.removeSelectedFilters({
-          group: filterStore.selectedGroupItem.vgroup,
-          groupItemName: filterStore.selectedGroupItem.name,
-          variant,
-        })
-
-        datasetStore.removeCondition({
-          subGroup: filterStore.selectedGroupItem.name,
-          itemName: variant[0],
-        })
-      }
-
-      datasetStore.fetchWsListAsync()
-    }
-
-    const variantsList: StatList[] = toJS(datasetStore.dsStat['stat-list'])
+    const statList: StatList[] = toJS(datasetStore.dsStat['stat-list']) ?? []
 
-    const currentGroup: StatList | undefined = variantsList.find(
+    const currentStatList: StatList | undefined = statList.find(
       (item: any) => item.name === filterStore.selectedGroupItem.name,
     )
 
+    const vgroup = filterStore.selectedGroupItem.vgroup
+    const groupItemName = filterStore.selectedGroupItem.name
+
+    const variants =
+      currentStatList?.variants?.map(
+        (variant: any): IVariantList => {
+          const variantQuantity =
+            filterStore.selectedFilters?.[`${vgroup}`]?.[`${groupItemName}`]?.[
+              `${variant[0]}`
+            ]
+
+          const isChecked = Boolean(variantQuantity)
+
+          return { variant, isChecked }
+        },
+      ) ?? []
+
+    const [variantList, setVariantList] = useState(variants)
+    const [currentPage, setCurrentPage] = useState(0)
+
+    const groupsPerPage = 20
+    const chunks = createChunks(variantList, groupsPerPage)
+
+    const handleSelect = (index: number, checked: boolean) => {
+      const localVariantList = [...variantList]
+
+      const currentIndex = currentPage * groupsPerPage + index
+
+      localVariantList[currentIndex].isChecked = checked
+
+      setVariantList(localVariantList)
+    }
+
+    const stringedVariants = JSON.stringify(variants)
+
+    useEffect(() => {
+      setVariantList(variants)
+
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [stringedVariants])
+
+    const handleClear = () => {
+      const localVariantList = variantList.map(element => {
+        element.isChecked = false
+
+        return element
+      })
+
+      setVariantList(localVariantList)
```

Comment:
I suppose, the selected filters should be cleared on the very left panel as well. At least, other filters behave like that.

Reply:
I agree. Fixed